### DISCUSSION TITLE
Matter build

### DISCUSSION THREAD

**Discussion by Cptmeme** [revk/xESP32-Faikout#571] [General]
Hi, 

A while ago I was very interested in a matter build for this project as I would love to be able to do away with my homebridge system. As you said back then that matter was not mature enough to have all the functions you needed to implement matter in this project, the matter implementation was understandably halted. 

Now I am a little further in my understanding of C programming than I was back then and wish to make a daikin controller based on your s21 protocol which supports matter. 

As I do not need most of the faikin features like the webpage and mqtt, only require basic control of my A/C and can somehow remember (I dont know whether this is correct as I cannot find it) you saying that you had a basic matter build. I was wondering whether this is actually the case and if so, if I could get access to it?

If this is not the case then I will have to base my implementation on the esp-matter air conditioning example. Would you have any tips for me regarding starting with this?

Kind regards!

**Comment by revk** [comment]
Matter has some serious issues.

* It seems to be specified by "use this code", and not by "here is the protocol definition" (tell me if I am wrong).
* The code seems to be C++, which is fine unless the entire code base with which you are using it is plain C. Possible, but a pain.
* Because the protocol is not clear, it is not clear what is needed, e.g. making a device that does MDNS and trying to use Matter did not work because matter uses MDNS as well and the code clashed in odd ways.
* You are expected to join a club and pay to certify every release of code, which is basically incompatible with rapid development releases, and even with basic open source. You can do as "test" but that always makes warnings.

This last point is what some hardware manufacturers are going for "Matter gateway", a device they can make and certify once, and then use with their devices as they release new hardware and software. It is an approach which I think will be the demise of Matter, which, otherwise, is very good.

Finally Home Assistant is easy, just works, runs a test matter gateway, and talks MQTT. Using Home Assistant should allow you to link Faikin to Matter without much problem.

Hope that makes sense.

**Reply by jinxlal** [reply]
I am trying to expose the Faikin from Home Assistant over Matter to Apple Home using the [Home Assistant Matter Hub Add-on](https://github.com/RiDDiX/home-assistant-matter-hub) but the controls in the Apple Home App are not correct.

This is how the controls look in Home Assistant which is correct as my AC does not go lower than 18 degrees Celsius and I have configured the same in the Faikin Web UI.

<img width="589" height="653" alt="image" src="https://github.com/user-attachments/assets/dff63e3d-96d0-4aec-b26c-e54c44bdf258" />

But in the Apple Home app it is not configured correctly and even the thermostat mode (heating, cooling, auto) does not stay in sync.

<img width="329" height="601" alt="image" src="https://github.com/user-attachments/assets/868b313a-8702-466f-b59a-9a2d714d61ac" />

Any ideas where the issue may lie?

**Reply by revk** [reply]
This sounds more like an HA issue than a Faikout issue...

**Reply by jinxlal** [reply]
Thanks, just posted in here in case some else who was trying to get the Faikout to work over Matter experienced something similar before. Either way, I will do some more digging on my end and revert if I find a solution.
